lib/ts/recipe/thirdparty/api/signinup.ts

```diff
@@ -108,9 +108,28 @@ export default async function signInUpAPI(recipeInstance: Recipe, req: Request,
     }
     let user = await recipeInstance.signInUp(provider.id, userInfo.id, emailInfo);
 
-    await recipeInstance.config.signInAndUpFeature.handlePostSignUpIn(user.user, accessTokenAPIResponse.data);
+    await recipeInstance.config.signInAndUpFeature.handlePostSignUpIn(
+        user.user,
+        accessTokenAPIResponse.data,
+        user.createdNewUser
+    );
 
-    await Session.createNewSession(res, user.user.id);
+    let action: "signup" | "signin" = user.createdNewUser ? "signup" : "signin";
+    let jwtPayloadPromise = recipeInstance.config.sessionFeature.setJwtPayload(
+        user.user,
+        accessTokenAPIResponse.data,
+        action
+    );
+    let sessionDataPromise = recipeInstance.config.sessionFeature.setSessionData(
+        user.user,
+        accessTokenAPIResponse.data,
+        action
+    );
+
+    let jwtPayload = await jwtPayloadPromise;
```

Comment:
See comment in emailpassword sign in / up API

Reply:
done